autoscale-source-kubernetes/src/main/java/com/github/autoscaler/source/kubernetes/K8sServiceSourceProvider.java

```diff
@@ -36,9 +33,9 @@ public ServiceSource getServiceSource(final ConfigurationSource configurationSou
     {
         try {
             final K8sAutoscaleConfiguration config = configurationSource.getConfiguration(K8sAutoscaleConfiguration.class);
-            Configuration.setDefaultApiClient(ClientBuilder.standard().build());
-            return new K8sServiceSource(config);
-        } catch (ConfigurationException | IOException e) {
+            final ApiClient apiClient = KubernetesClientFactory.createClientWithCertAndToken();
+            return new K8sServiceSource(config, apiClient);
+        } catch (final Exception e) {
```

Comment:
I assume catching unchecked exceptions is not intended, so change this to:

```suggestion
        } catch (final ConfigurationException | FailedToCreateKubernetesClientException e) {
```

If catching unchecked exceptions is intentional then be explicit about it:

```suggestion
        } catch (final ConfigurationException | FailedToCreateKubernetesClientException | RuntimeException e) {
```